components/BarcodeScanner.vue

```diff
@@ -0,0 +1,334 @@
+<template>
+  <div
+    ref="videoContainer"
+    class="videoContainer row justify-center items-center">
+    <!-- Spinners -->
+    <div class="full-height row items-center">
+      <slot
+        v-if="loadingCamera"
+        name="cameraSpinner">
+        <Spinner />
+      </slot>
+      <slot
+        v-if="scanning"
+        name="scannerSpinner">
+        <Spinner />
+      </slot>
+    </div>
+    <!-- Video Stream -->
+    <video
+      class="dbrScanner-video"
+      playsinline="true" />
+    <!-- Guide -->
+    <canvas
+      v-show="!loadingCamera && !scanning"
+      id="guide"
+      style="position: absolute; top:0; height: 100%; width: 100%" />
+    <!-- Tip Text -->
+    <div
+      v-if="!loadingCamera && !scanning"
+      class="text-white"
+      style="position: absolute;"
+      :style="`bottom: calc(${regionTop}% - 30px)`">
+      {{tipText}}
+    </div>
+    <!-- Close -->
+    <q-btn
+      v-if="!loadingCamera && !scanning"
+      flat
+      fab
+      :ripple="false"
+      size="16px"
+      color="white"
+      icon="fas fa-times"
+      class="q-ma-sm close-btn"
+      @click.native="close()" />
+    <!-- Buttons -->
+    <Buttons
+      :camera-list="cameraList"
+      @updateCamera="updateCamera($event)"
+      @upload="upload($event)" />
+  </div>
+</template>
+
+<script>
+/*!
+  * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+  */
+import DBR from '../helpers/dbr.js';
+import driverLicenseFields from '../helpers/driverLicenseFields.js';
+import Buttons from './Buttons.vue';
+import Spinner from './Spinner.vue';
+
+DBR.BarcodeReader.license = '100902848';
+
+export default {
+  name: 'BarcodeScanner',
+  components: {
+    Buttons,
+    Spinner
+  },
+  props: {
+    tipText: {
+      type: String,
+      default: 'Scan barcode on the back of license.'
+    },
+    guideColor: {
+      type: String,
+      default: 'rgb(254, 142, 20)'
+    }
+  },
+  data() {
+    return {
+      loadingCamera: true,
+      scanning: false,
+      isDestroyed: false,
+      scanner: null,
+      reader: null,
+      dlInfo: null,
+      clientHeight: null,
+      clientWidth: null,
+      regionScale: 0.4,
+      cameraList: [],
+      currentCamera: null,
+      maskCanvas: null,
+      maskCanvasContext: null,
+    };
+  },
+  computed: {
+    regionMaskEdgeLength() {
+      const regionMaskEdgeLength =
+        this.regionScale * Math.min(this.clientWidth, this.clientHeight);
+      return Math.floor(regionMaskEdgeLength);
+    },
+    regionLeft() {
+      const left =
+        (this.clientWidth - this.regionMaskEdgeLength) / 2 / this.clientWidth;
+      if(this.clientWidth > this.clientHeight) {
+        return Math.round(left * 100) - 25;
```

Comment:
Would probably be great to put the raw numbers into constants - like, what does 100 stand for, etc.

Reply:
This is math that I took from their barcode reader example to ensure that our reader does the same thing as theirs. In this case the 100 is a percent.

Reply:
A comment could explain the units here.